src/structures/message.ts

```diff
@@ -1,36 +1,215 @@
+import { Member } from "./member.ts";
+import { Role } from "./role.ts";
+
+const baseMessage: Partial<Message> = {
+  get guild() {
+    return cache.guilds.get(this.guildID);
+  },
+  get member() {
+    return cache.members.get(this.author?.id);
+  },
+  get guildMember() {
+    return this.member?.guilds.get(this.guildID);
+  },
+  get link() {
+    return `https://discord.com/channels/${this.guildID ||
+      "@me"}/${this.channelID}/${this.id}`;
+  },
+  get mentionedRoles() {
+    return this.mentionRoleIDs?.map(id => this.guild?.roles.get(id)) || [];
+  },
+  get mentionedChannels() {
+    return this.mentionChannelIDs?.map(id => cache.channels.get(id)) || [];
+  },
+  get mentionedMembers() {
+    return this.mentions?.map(id => cache.members.get(id)) || []
+  },
+
+  // METHODS
+  delete(reason, delayMilliseconds) {
+    return deleteMessageByID(this.channelID!, this.id!, reason, delayMilliseconds);
+  },
+  edit(content) {
+    return editMessage(this, content);
+  },
+  pin() {
+    return pin(this.channelID!, this.id!)
+  },
+  addReaction(reaction) {
+    return addReaction(this.channelID!, this.id!, reaction);
+  },
+  addReactions(reactions, ordered) {
+    return addReactions(this.channelID!, this.id!, reactions, ordered);
+  },
+  sendResponse(content) {
+    const contentWithMention = typeof content === "string"
+      ? { content, mentions: { repliedUser: true }, replyMessageID: this.id }
+      : {
+        ...content,
+        mentions: { ...(content.mentions || {}), repliedUser: true },
+        replyMessageID: this.id,
+      };
+  
+    return sendMessage(this.channelID!, contentWithMention);
+  },
+  reply(content) {
+    return sendMessage(this.channelID!, content);
+  },
+  alert(content, timeout = 10, reason = "") {
+    return sendMessage(this.channelID!, content).then(response => response.delete(reason, timeout * 1000).catch(console.error))
+  },
+  alertResponse(content, timeout = 10, reason = "") {
+    return this.sendResponse!(content).then(response => response.delete(reason, timeout * 1000).catch(console.error))
+  },
+  removeAllReactions() {
+    return removeAllReactions(this.channelID!, this.id!);
+  },
+  removeReactionEmoji(reaction) {
+    return removeReactionEmoji(this.channelID!, this.id!, reaction);
+  },
+  removeReaction(reaction) {
+    return removeReaction(this.channelID!, this.id!, reaction);
+  },
+};
 
 export async function createMessage(data: MessageCreateOptions) {
   const {
     guild_id: guildID,
     channel_id: channelID,
     mentions_everyone: mentionsEveryone,
-    mention_channels: mentionChannels,
-    mention_roles: mentionRoles,
+    mention_channels: mentionChannelIDs,
+    mention_roles: mentionRoleIDs,
     webhook_id: webhookID,
     message_reference: messageReference,
     edited_timestamp: editedTimestamp,
     referenced_message: referencedMessageID,
     ...rest
   } = data;
 
-  const message = {
-    ...rest,
+  const restProps = {};
```

Comment:
```suggestion
  const restProps: Record<keyof MessageCreateOptions, any> = {};
```
`any` is the return type of createNewProp(), you may change if you would like to.